rs/bitcoin/ckbtc/minter/src/storage.rs

```diff
@@ -68,7 +70,16 @@ fn encode_event(event: &Event) -> Vec<u8> {
 ///
 /// This function panics if the event decoding fails.
 fn decode_event(buf: &[u8]) -> Event {
-    ciborium::de::from_reader(buf).expect("failed to decode a minter event")
+    #[derive(Deserialize)]
+    #[serde(untagged)]
```

Comment:
Is this the reason why backward compatibility is preserved (namely, old `Event` can still be decoded as `SerializedEvent`)? If so, perhaps you can add some comment to explain why it is important to have this annotation.

Also, I think it needs to be explained that why decoding the new `Event` won't be mistaken as `SerializedEvent::Legacy` (or even when it does, it doesn't matter).
 
I suggest moving the `enum` definition to top level so that you can add some test cases.

Reply:
Thanks for the input! I actually think it makes sense to keep the `enum` inside the function definition since it is actually an implementation detail, but I agree that some test cases make sense. I added a comment explaining how this works and some unit tests to verify that both types of events can be encoded/decoded.

Reply:
LGTM! Thanks for adding the test case